### PR TITLE
Add iterator support to pyknowhere interface.

### PR DESCRIPTION
**Pull request by zhengbuqian**
Add GetAnnIterator method to the swigknowhere IndexWrap class to wrap the AnnIterator method, so iterator can also be used in pyknowhere.

/kind improvement

### CODE REVIEW

File: python/knowhere/knowhere.i [hunk 1]
```diff
@@ -113,6 +115,25 @@ public:
     PyThreadState* save;
 };
 
+class AnnIterator {
```
Comment (cydrain) [inline python/knowhere/knowhere.i:118]: Suggest to rename to "IteratorWrap"

File: python/knowhere/knowhere.i [hunk 2]
```diff
@@ -157,6 +178,22 @@ class IndexWrap {
         }
     }
 
+    std::vector<AnnIterator>
+    GetAnnIterator(knowhere::DataSetPtr dataset, const std::string& json, const knowhere::BitsetView& bitset, knowhere::Status& status) {
```
Comment (cydrain) [inline python/knowhere/knowhere.i:182]: Rename to "GetIterator"

File: python/knowhere/knowhere.i (hunk 1 above)
Comment (zhengbuqian) [reply python/knowhere/knowhere.i]: Done

File: python/knowhere/knowhere.i (hunk 2 above)
Comment (zhengbuqian) [reply python/knowhere/knowhere.i:182]: Keeping keyword `Ann` as discussed.

File: python/knowhere/knowhere.i [hunk 3]
```diff
@@ -113,6 +115,25 @@ public:
     PyThreadState* save;
 };
 
+class AnnIteratorWrap {
+ public:
+    AnnIteratorWrap(std::shared_ptr<IndexNode::iterator> it = nullptr) : it_(it) {
```
Comment (cydrain) [inline python/knowhere/knowhere.i:120]: suggest assert here if it is NULL

**Pull request merged**